torch_xla/experimental/pallas_kernels/quantized_matmul_kernel.py

```diff
@@ -216,6 +205,40 @@ def quantized_matmul_int8(
   # For best performance, only enable this behavior when single input block is used per batch.
   save_q_x = quantize_activation and n_in == 1 and n_out > 1
 
+  acc_dtype = jnp.int32 if quantize_activation else jnp.float32
+
+  # Calculate in/out VMEM size.
+  x_size = batch_block_size * in_block_size * x.dtype.itemsize
+  x_abs_max_val_size = batch_block_size * x_abs_max_val.dtype.itemsize
+  w_size = out_block_size * in_block_size * w.dtype.itemsize
+  scalar_size = out_block_size * scalar.dtype.itemsize
+  out_size = batch_block_size * out_block_size * x.dtype.itemsize
+
+  vmem_in_out = x_size + x_abs_max_val_size + w_size + scalar_size + out_size
+  vmem_in_out *= 2  # Account for compute and vreg spills.
+
+  # Account for double buffering.
+  # Double buffering is used only if there are multiple blocks per in/out.
+  vmem_in_out += x_size if (n_bs > 1 or n_in > 1) else 0
+  vmem_in_out += x_abs_max_val_size if (n_bs > 1) else 0
+  vmem_in_out += w_size if (n_out > 1 or n_in > 1) else 0
+  vmem_in_out += scalar_size if (n_out > 1) else 0
+  vmem_in_out += out_size if (n_bs > 1 or n_out > 1) else 0
+
+  # Calculate scratch VMEM size.
+  acc_size = batch_block_size * out_block_size * 4
```

Comment:
could you replace the 4 with something like jnp.type(acc_dtype).itemsize?

Reply:
similar to the 1 and 4 in the below 2 lines

Reply:
Done.